### ISSUE TITLE
[History and sync status] Improve menu label and make screen heading consistent

### ISSUE THREAD

**Issue by gino-m** [google/ground-android#2625]
"History and sync status" is impossibly long.. can we shorten this to "Upload status"? We don't show when remote changes are downloaded, so that might be more clear. @amegantz ?

**Comment by amegantz** [comment]
"Upload status" is ambiguous, as "upload" could be interpreted as a noun or a verb ("Do I need to upload my status?"), so maybe:
- Survey status
- Data status
- Survey upload status

WDYT?

![image](https://github.com/user-attachments/assets/e8abefce-c98a-4277-aca1-978c0675485c)

**Comment by gino-m** [comment]
"Survey status" implies progress in the overall survey. I don't object to the word "sync" provided it's used consistently. How about "Data sync status"?

**Comment by amegantz** [comment]
"Data sync status" works for me!
